packages/eslint-plugin-gestalt/src/prefer-flex.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @fileoverview Encourage the use of Flex instead of Box
+ *
+ * We prefer using Flex over Box to better separate concerns and to encourage
+ * the usage of `gap`. This linter checks for usage of Box that could be Flex
+ * given the used props.
+ */
+
+// @flow strict
+import { buildProps } from './eslintASTHelpers.js';
+import { renameTagWithPropsFixer, updateGestaltImportFixer } from './eslintASTFixers.js';
+
+const sharedProps = [
+  'alignContent',
+  'alignItems',
+  'alignSelf',
+  'children',
+  'direction',
+  'flex',
+  'height',
+  'justifyContent',
+  'maxHeight',
+  'maxWidth',
+  'minHeight',
+  'minWidth',
+  'overflow',
+  'width',
+  'wrap',
+];
+
+export const errorMessage =
+  "Please use Flex for flexbox layouts. If you are wrapping children in Boxes to set margin/padding, try using Flex's `gap` prop instead!";
+
+const rule = {
```

Comment:
can we match the patterns re Flow and metadata from the other files? Gestalt restrictions


```
import { type ESLintRule } from './eslintFlowTypes.js';

const rule: ESLintRule = {
  meta: {
    type: 'suggestion',
    docs: {
      description:
      category: 'Gestalt restrictions',
      recommended: true,
      url: 'https://gestalt.pinterest.systems/Eslint%20Plugin#gestaltprefer-flex',
    },
    fixable: 'code',
    schema: ([]: $ReadOnlyArray<empty>),
    messages: {
      preferFlex: "Please use Flex for flexbox layouts. If you are wrapping children in Boxes to set margin/padding, try using Flex's `gap` prop instead!",
    },
  },

```